isamples_metadata/taxonomy/metadata_models.py

```diff
@@ -167,40 +172,40 @@ def classify_by_rule(self, text: str, description_map: dict) -> Optional[str]:
         If the record corresponds to a rule, returns the rule-defined label
         Else return None
         """
-        # 1. check if record does not have enough information
-        # if not enough information given,
-        # give "Material" label
-        if not self.check_informative(text, description_map):
-            return "Material"
-
-        # 2. rule-based classification
+        # 1. rule-based classification
         # extract fields that we need to consider for the rules
         fields_to_check = [
-            "sampleType",
+            "supplementMetadata_sampleType",
             "cruiseFieldPrgrm",
             "igsnPrefix",
             "description",
-            "primaryLocationType"
+            "supplementMetadata_primaryLocationType"
         ]
         # build a map that stores the fields that we are interested
         field_to_value = collections.defaultdict(str)
         for key, value in description_map.items():
             if key in fields_to_check:
+                # remove nested label
+                key = key.replace("supplementMetadata_", "")
                 field_to_value[key] = value
 
         # check if record is invalid
         # i.e., not a sample
         if self.check_invalid(field_to_value):
             return "Invalid"
-
         # check if the fields fall into the rules
         # if it does not, return None
         result = self.classify_by_sample_type(field_to_value)
         if result:
             # map to controlled vocabulary
             return SESARClassifierInput.source_to_CV[result]
         else:
-            return None
+            # check if there is enough information to do classification
+            # give "Material" label
+            if not self.check_informative(text, description_map):
```

Comment:
Fix the rule-based classification script, by basically re-ordering the steps. We want to check if the record is informative enough to conduct machine learning based classification before we pass it to the model. Previously the `check_informative` was the first step of rule-based classification, but this misses the cases when we can actually get the label from `classify_by_sample_type`